src/pages/home/search/SearchModal.tsx

```diff
@@ -32,12 +50,38 @@ function SearchModal() {
     }
   };
 
-  const onSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
-    const recentSearch = [...savedSearch];
-    // currently it is saving each word is typed, need to handle with debounce
-    recentSearch.push(event.target.value);
-    setSavedSearch(recentSearch);
+  const onInputValueChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value);
+    setCurrHoveredOption(undefined);
+  };
+
+  const onSearch = (newSearch: string, doNavigate = true) => {
+    setSavedSearch((prev) => {
+      const updatedSearch = [...prev, newSearch];
+      const uniqueSearch = [...new Set(updatedSearch)];
```

Comment:
Can we use debounceInputValue to save in localStorage as well apart from using in autoComplete? It will help to store complete user search in localStorage

Reply:
Ok, I thought each every type is getting saved in localStorage ,mybad